internal/errorhandling/safe.go

```diff
@@ -0,0 +1,36 @@
+package errorhandling
+
+import "fmt"
+
+// safe2 runs the specified function and returns its result value or returned error. If a panic occurs, it returns the
```

Comment:
I don't quite understand the value in this either --- all the error conditions that AES-GCM panics on, could be handled by us, the caller, and thus assert they won't panic.

E.g,. https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/crypto/cipher/gcm.go;l=205 can be caught [by checking against `AEAD.NonceSize()`](https://pkg.go.dev/crypto/cipher#AEAD) and the inexact buffer overlaps can be proven not to occur if we're decrypting to a different buffer (guaranteed by passing `nil` as `dst` like we're doing). Message length checks should also probably be done here, as I don't think we want to be encrypting 4GB of data, generally speaking, and probably want a much lower threshold.

Aside: I suspect any panics observed were during development and shouldn't apply to productions (and even then, should be caught by explicit safeguards in the code here and not passing unchecked values straight to the stdlib).

My 2c., but I think this can be reworked to avoid the need for this entirely, in this code.

Reply:
Thank you @cipherboy  and @serdardalgic ! My thinking when implementing this was that the authors of the crypto package seem to think that it's a perfectly fine thing to panic in the face of an invalid input. Panics should, according to a lot of opinions, only be used for unrecoverable errors, which an invalid input clearly isn't. I don't trust the authors of the crypto package to not add panics in the future when they want to handle another case. I am not opposed to preemptively checking the input, but I don't want to run into a situation when in a future Go version, this starts crashing for some reason.